File: 463-island-perimeter/island-perimeter.java
class Solution {
    public int count=0;
    public void dfs(int [][]arr,int i,int j){
        if(i<0||j<0||i>=arr.length||j>=arr[0].length||arr[i][j]==0){
            count++;
            return ;
        }
        if(arr[i][j]==2){
            return ;
        }
        arr[i][j]=2;
        dfs(arr,i+1,j);
        dfs(arr,i-1,j);
        dfs(arr,i,j+1);
        dfs(arr,i,j-1);

    }
    public int islandPerimeter(int[][] grid) {
        for(int i=0;i<grid.length;i++){
            for(int j=0;j<grid[0].length;j++){
                if(grid[i][j]==1){
                    dfs(grid,i,j);
                    return count;
                }
            }
        }
        return 0;
    }
}

File: 463-island-perimeter/island-perimeter.cpp
class Solution {
public:
    int cnt=0;
    void dfs(vector<vector<int>> &grid,int r,int c){
        if(r<0 || c<0 || r>=grid.size()||c>=grid[0].size()||grid[r][c]==0){
            cnt++;
            return;
        }
       
        if(grid[r][c]==2)
            return;
            grid[r][c]=2;
            dfs(grid,r+1,c);
            dfs(grid,r-1,c);

            dfs(grid,r,c-1);

            dfs(grid,r,c+1);

        
    }
    int islandPerimeter(vector<vector<int>>& grid) {
        for(int r=0;r<grid.size();r++){
            for(int c=0;c<grid[0].size();c++){
                if(grid[r][c]==1){
                    dfs(grid,r,c);
                    return cnt;
                }
            }
        }
        return 0;
    }
};